jitbuilder/build/rules/gnu/common.mk

```diff
@@ -63,7 +63,7 @@ $(call RULE.cpp,$(JIT_PRODUCT_BUILDNAME_OBJ),$(JIT_PRODUCT_BUILDNAME_SRC))
 .phony: $(JIT_PRODUCT_BUILDNAME_SRC)
```

Comment:
I'm guessing that it was specifically this phony rule which generates the version unconditionally that was the problem?  I think that `add_custom_command()` should have been working fine, is that correct?

Reply:
I think that should be `.PHONY`, not `.phony`.

The main problem was that (when the rule ran), it would modify the target unconditionally. The new version script is given a chance to read the target file so it can be left untouched when the contents are correct.

Assuming the cmake custom command ran at the right times (I'm not sure that's the case, but I'd prefer to leave that to a separate PR), it would also unconditionally modify the file.